cpp/src/arrow/flight/client.cc

```diff
@@ -273,7 +267,29 @@ class GrpcClientInterceptorAdapter : public grpc::experimental::Interceptor {
   }
 
  private:
+  void ReceivedHeaders(
+      const std::multimap<grpc::string_ref, grpc::string_ref>& metadata) {
+    if (received_headers_) {
+      return;
+    }
+    received_headers_ = true;
+    CallHeaders headers;
+    for (const auto& entry : metadata) {
+      headers.insert({util::string_view(entry.first.data(), entry.first.length()),
+                      util::string_view(entry.second.data(), entry.second.length())});
+    }
+    for (const auto& middleware : middleware_) {
+      middleware->ReceivedHeaders(headers);
```

Comment:
What is the semantics of `ReceivedHeaders` if headers are empty? Is it always called? Only in some circumstances?